### PR TITLE
Transfer also log.Sink between contexts in routine pkg

### PR DESCRIPTION
**Pull request by Smuerdt**
Related to https://github.com/pace/bricks/issues/179

### CODE REVIEW

**Comment by Teelevision** [comment]
Please fix failed checks.

File: maintenance/log/sink.go [hunk 1]
```diff
@@ -29,6 +29,18 @@ func SinkFromContext(ctx context.Context) (*Sink, bool) {
 	return sink, ok
 }
 
+// ContextTransfer gets the logger and sink from the sourceCtx
+// and returns a new context based on targetCtx and these two.
+// If there is no sink in sourceCtx, a new one is created.
+func ContextTransfer(sourceCtx, targetCtx context.Context) context.Context {
```
Comment (Teelevision) [inline maintenance/log/sink.go]: Why would you want to create a sink if there isn't one in the original context?

File: maintenance/log/sink.go [hunk 2]
```diff
@@ -29,6 +29,18 @@ func SinkFromContext(ctx context.Context) (*Sink, bool) {
...
+	sink, ok := SinkFromContext(sourceCtx)
+	if !ok {
+		sink = new(Sink)
+	}
+
+	return ContextWithSink(targetCtx, sink)
+}
+
```
Comment (Teelevision) [inline maintenance/log/sink.go]: It doesn't actually transfer the logger. It only transfers the sink. It should transfer both.

Comment (Smuerdt) [reply maintenance/log/sink.go]: It does in `ContextWithSInk` but doesn't matter I changed it.

File: pkg/routine/routine_test.go [hunk 3]
```diff
@@ -33,13 +32,13 @@ func TestRun_createsNewContext(t *testing.T) {
...
-	ctx := logger.WithContext(context.Background())
+	var sink log.Sink
+	logger := log.Logger()
+	ctx := log.ContextWithSink(logger.WithContext(context.Background()), &sink)
 	waitForRun(ctx, func(ctx context.Context) {
 		log.Ctx(ctx).Debug().Msg("foobar")
 	})
-	require.Contains(t, buf.String(), "foobar")
+	require.Contains(t, string(sink.ToJSON()), "foobar")
 }
```
Comment (Teelevision) [inline pkg/routine/routine_test.go]: Please don't re-purpose this test. This test did test that the logger is transferred. Add an additional test to test that the sink is transferred, please. The sink cannot go without the logger, but the logger can go without the sink.

Comment (Smuerdt) [reply pkg/routine/routine_test.go]: done

**Pull request merged**